### PR TITLE
workflow: sign release containers

### PR DESCRIPTION
**Pull request by tkatila**
Signs release containers with `cosign`. Also modified the publish workflow to support other registries. Makes it easier to test changes.

Tests:
https://hub.docker.com/repository/docker/tkatila/intel-gpu-plugin/tags

Test runs:
https://github.com/tkatila/intel-device-plugins-for-kubernetes/actions/runs/10367001757/job/28697484565
https://github.com/tkatila/intel-device-plugins-for-kubernetes/actions/runs/10367083492/job/28697736086

cosign can be used to verify containers:
`cosign verify --certificate-oidc-issuer https://token.actions.githubusercontent.com --certificate-identity-regexp https://github.com/tkatila/intel-device-plugins-for-kubernetes/.github/workflows/lib-publish.yaml.* tkatila/intel-gpu-plugin:0.30.101  | jq .`

### CODE REVIEW

**Review by mythi — APPROVED** [review]
nice work! I think we also need to document this in our top-level `README` so that users know to take it in use

**Comment by tkatila** [comment]
> nice work! I think we also need to document this in our top-level `README` so that users know to take it in use

Yep, it did cross my mind also. I'll add that.

File: README.md [hunk 1]
```diff
@@ -291,6 +292,12 @@ the branches and releases in this repository.
 [imagePullPolicy](https://kubernetes.io/docs/concepts/containers/images/#updating-images)
 ```IfNotPresent``` and can be changed with ```scripts/set-image-pull-policy.sh```.
 
+### Signed container images
+
+Beginning of 0.31 release, release images (`0.31.0` etc., not `devel`) will be signed with keyless signing using `cosing`. The signing proof is stored in [rekor.sigstore.dev](https://rekor.sigstore.dev) in an append-only transparency log. The signature is also stored within the dockerhub.
```
Comment (mythi) [inline README.md:297]: small edits perhaps:
```suggestion
Starting from 0.31 release, the images (`0.31.0` etc., not `devel`) are signed with keyless signing using `cosing`. The signing proof is stored in [rekor.sigstore.dev](https://rekor.sigstore.dev) in an append-only transparency log. The signature is also stored within the dockerhub.
```

**Review by mythi — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -291,6 +292,12 @@ the branches and releases in this repository.
 [imagePullPolicy](https://kubernetes.io/docs/concepts/containers/images/#updating-images)
 ```IfNotPresent``` and can be changed with ```scripts/set-image-pull-policy.sh```.
 
+### Signed container images
+
+Starting from 0.31 release, the images (`0.31.0` etc., not `devel`) are signed with keyless signing using `cosing`. The signing proof is stored in [rekor.sigstore.dev](https://rekor.sigstore.dev) in an append-only transparency log. The signature is also stored within the dockerhub.
```
Comment (eero-t) [inline README.md:297]:
```suggestion
Starting from 0.31 release, the images (`0.31.0` etc., not `devel`) are signed with keyless signing using `cosign`. The signing proof is stored in [rekor.sigstore.dev](https://rekor.sigstore.dev) in an append-only transparency log. The signature is also stored within the dockerhub.
```

Comment (tkatila) [reply README.md:297]: ah, yes, let's all co-sing some songs.

**Pull request merged**